cmd/snapctl/task.go

```diff
@@ -114,14 +114,226 @@ func stringValToInt(val string) (int, error) {
 	return int(parsedField), nil
 }
 
+func (t *task) checkScheduleWindow(start time.Time, stop time.Time) error {
+	// if the stop time defined by the user is in the past, then throw
+	// an error (at least some of the defined window should be in the future)
+	if time.Now().Add(createTaskNowPad).After(stop) {
+		return fmt.Errorf("Usage error (poorly defined time window); when constructing a new 'windowed' schedule, part of that window must be in the future")
+	} else if start.After(stop) {
+		return fmt.Errorf("Usage error (poorly defined time window); the start date-time (%v) must be before the stop date-time (%v)", start, stop)
+	}
+	// return a nil error (indicating success)
+	return nil
+}
+
+// Parses the command-line parameters and uses them to override the underlying schedule
+// in this task (if any) or set a schedule for that task (if one is not already defined)
+func (t *task) setWindowedSchedule(start *time.Time, stop *time.Time, duration *time.Duration) error {
+	// if this task does not have an existing schedule, create an empty
+	// windowed schedule
+	if t.Schedule == nil {
```

Comment:
t.Schedule should never be nil here. A task manifest should always include a schedule, and if it doesn't, an error should be thrown long before the code gets down this path.